File: python/min-cost-climbing-stairs.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        current_minimum_cost = 0
        candidate_1 = cost[0]
        candidate_2 = cost[1]
        for i in range(2, len(cost)):
            current_minimum_cost = cost[i] + min(candidate_1, candidate_2)
            candidate_1 = candidate_2
            candidate_2 = current_minimum_cost
        return min(candidate_1, candidate_2)

File: rust/src/min_cost_climbing_stairs.rs
impl Solution {
    fn min(m: i32, n: i32) -> i32 {
        if m > n {
            return n;
        }
        return m;
    }
    pub fn min_cost_climbing_stairs(cost: Vec<i32>) -> i32 {
        let mut current_minimum_cost = 0;
        let mut candidate_1 = cost[0];
        let mut candidate_2 = cost[1];
        for i in 2..cost.len() {
            current_minimum_cost = cost[i] + Solution::min(candidate_1, candidate_2);
            candidate_1 = candidate_2;
            candidate_2 = current_minimum_cost;
        }
        return Solution::min(candidate_1, candidate_2);
    }
}
